core/src/main/java/io/aiven/kafka/tieredstorage/config/RemoteStorageManagerConfig.java

```diff
@@ -95,6 +97,11 @@ public class RemoteStorageManagerConfig extends AbstractConfig {
         + "Remote Log Segment metadata on Remote Log Metadata Manager back-end. "
         + "Allowed values: " + Arrays.toString(SegmentCustomMetadataField.names());
 
+    private static final String UPLOAD_RATE_LIMIT_BYTES_CONFIG = "upload.rate.limit.bytes.per.second";
+    private static final String UPLOAD_RATE_LIMIT_BYTES_DOC = "Upper bound on bytes to upload "
+        + "(therefore read from disk) per second. This affects only the upload rate. "
+        + "Rate limit must be equal or larger than chunk.size to allow a proper rate per second.";
```

Comment:
Just curious: is this some limitation on the Bucket4j side? Like can't count 1/2?

Reply:
We added this additional lower bound to avoid upload rates that are "too low" (e.g. less than a chunk upload per second) -- see https://github.com/Aiven-Open/tiered-storage-for-apache-kafka/pull/548#discussion_r1609522472
Feels like a reasonable step now but wonder if users with large chunk sizes would be happy with this. Maybe worth removing it.